lib/ipmi_hpmfwupg.c

```diff
@@ -1398,6 +1398,7 @@ HpmfwupgGetBufferFromFile(char *imageFilename,
 	if (ret != 0) {
 		lprintf(LOG_ERR, "Failed to seek in the image file '%s'",
 				imageFilename);
+		fclose(pImageFile);
```

Comment:
I'd prefer a Linux kernel style [centralized exiting](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#centralized-exiting-of-functions).
Also, I'd set `rc` to `HPMFWUPG_ERROR` by default and only set it to `HPMFWUPG_SUCCESS` when we're right before the end of the successful execution path. That is usually right before that centralized exit point.